src/Reflection/Mutation/AddClassProperty.php

```diff
@@ -0,0 +1,53 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Reflection\Mutation;
+
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\Stmt\PropertyProperty;
+use ReflectionProperty as CoreReflectionProperty;
+use Roave\BetterReflection\Reflection\Mutator\ReflectionClassMutator;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+
+class AddClassProperty
+{
+    /**
+     * @var ReflectionClassMutator
+     */
+    private $mutator;
+
+    public function __construct()
+    {
+        $this->mutator = new ReflectionClassMutator();
+    }
+
+    public function __invoke(
+        ReflectionClass $classReflection,
+        string $propertyName,
+        int $visibility = CoreReflectionProperty::IS_PUBLIC,
```

Comment:
Would suggest removing default values overall

Reply:
Perfect, I was thinking about it but I implemented the same way that already exist at the end.